chef/cookbooks/network/templates/default/ovs-pre-up.sh.erb

```diff
@@ -12,3 +12,17 @@ ovs-vsctl del-fail-mode <%= @bridgename %>
 <%
   end
 -%>
+
+# Open vSwitch older than 2.11 is starting 2*$(nproc)*3/4 netlink sockets per
+# number of cpu cores and per port. On machines with high core
+# count (e.g. 56 or higher) this can quickly exceed the total fd
+# limit set by ovs on itself on startup of 65536.
+
+# By Limit the handler threads to 8 per port we can handle ~ 4000 ports
+# instead of just ~ 700.
+# see https://mail.openvswitch.org/pipermail/ovs-dev/2018-September/352402.html
+# see bsc#1110865
+max_handler_threads=8
+if [ $(nproc) -gt $max_handler_threads ]; then
+    ovs-vsctl set Open_vSwitch $(ovs-vsctl show | head -n 1) other_config:n-handler-threads=$max_handler_threads
```

Comment:
I'd prefer to take an approach as in line 4 instead of relying on bash commands

Reply:
Any idea how to retrieve the Open_vSwitch id other than show | head -n 1 ? I tried several things, and was even desparate enough to grep the documentation and couldn't find a hint.

Reply:
@dirkmueller 
`ovs-vsctl get Open_vSwitch . _uuid`

Reply:
didn't want to nitcpick sorry, just thought that there was something like `<%= @datapath_id %>` already that we can leverage

Reply:
I think you could just use

`ovs-vsctl set Open_vSwitch . other_config:n-handler-threads=$max_handler_threads`